swap_meet/vendor.py

```diff
@@ -1,2 +1,60 @@
+from .item import Item
 class Vendor:
-    pass
+    def __init__(self, inventory = None ):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, inventory_item):
+        self.inventory.append(inventory_item)
+        return inventory_item
+
+    def remove(self, inventory_item):
+        if inventory_item in self.inventory:
+            self.inventory.remove(inventory_item)
+            return inventory_item
+        else:
+            return False
+    def get_by_category(self, category):
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items 
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            other_vendor.add(my_item)
+            self.remove(my_item)
+            self.add(their_item)
+            other_vendor.remove(their_item)
+            return True
+        else:
+            return False 
+
+    def swap_first_item(self, other_vendor):
+        if self.inventory == [] or other_vendor.inventory == []:
+            return False
+        else:
```

Comment:
It's a little more pythonic to check for an empty list by taking advantage of its falsiness rather than explicitly checking the `len()`, as in

```python
        if not self.inventory or not other_vendor.inventory:
            return False
```

Also, notice that since the only thing you're doing in the positive condition is to `return`, this is essentially a guard clause. We can leave off the `else` and unindent the lines that were within it.